### ISSUE TITLE
Changing noted

### ISSUE THREAD

**Issue by ghost** [Terodom/termNote#8]
Making it so that noted checks for the date and time

**Comment by ghost** [comment]
Maybe make it so that it sees the syntax` %YYYY.%MM.%DD %HH:%MM`

**Comment by balsoft** [comment]
That would be nice. Actually, why not make it modular? E.g. allow to specify date/time format in configuration?

**Comment by ghost** [comment]
That was actually my plan, i will implement libconfig

**Comment by balsoft** [comment]
I actually have some suggestions for noted.
First of all, there should be a configuration option for date/time formats.
Secondly, I believe a "strictness" option would be nice.
I suggest a following set of rules for "soft" mode:
- Don't notify when no part of date/time is found in a note;
- Parse dates and times separately from each other. 
- Different parts of date should only be parsed together and notification should toggle when all of them match;
- if there are two overlapping definitions of same part of date or time, we should notify when either one matches;
- When time is specified (HH:MM), match only that time;
- When hour is specified in AM/PM format, assume minutes are 00;
- When no time is specified, assume 00:00
- When a date is specified (DD.MM or according to format), match that date only; 
- When a weekday is specified, match days which are that weekday;
- When no weekday is specified, match any day;
- When a month name is specified, match every day of that month;
- When no month name is specified, match any month;
- When a year is specified, match every day of the year;
- When no year is specified, match any year;
- When no date is specified at all, match any day.
Examples:
`call Mom 6PM` - remind every day at 18:00\
`call Mom 18:00` - the same\
`call Mom Friday 6pm` - every friday at 18:00\
`call Mom Friday Saturday 6PM` - every Friday and every saturday at 18:00\
`call Mom 20.01 6PM` - only on January 20th every year at 18:00\
`call Mom January 6PM` - every day of January at 18:00\
`call Mom January 10AM 6PM` - every day of January at 10AM and at 6PM
`call Mom January Friday 12:34` - every friday of January at 12:34\
`call Mom January Friday 15.05 18.05 12:34` - Every friday of January (as they are part of the same date specification), may 15th and may 18th, All of those at 12:34;\
`call Mom 20.01.2018 18:00` - only once

And strict mode should just allow the last option (full date and time, defined once)
While that would be quite hard to implement, I think it would be just perfect for my needs! (And perhaps it should fit most workflows). For now, just matching time is a good-enough solution for me. (I am already using termNote in my workflow to add quick reminders :+1: )

**Comment by ghost** [comment]
I've been looking at libconfig for the config system, but it doesn't do well with multithreading

**Comment by balsoft** [comment]
What do you mean exactly? As I see it, neither termNote nor noted should write to the configuration file, only read it. What's the problem then?

**Comment by balsoft** [comment]
Maybe I can implement my "soft" mode before config system then?

**Comment by ghost** [comment]
Sure, do that

**Comment by ghost** [comment]
Will implement the config system this weekend, i've been sick the whole week, that's why i only maintained and didn't contribute myself

**Comment by balsoft** [comment]
Soft mode is coming, I am going to test it myself for some time and then PR it. I haven't quite followed my own rules, so it needs additional checking.

**Comment by balsoft** [comment]
I think this can be closed and if there is an issue with parsing algo, a new issue should be opened.
